packages/redux/src/checkout/draftOrders/reducer.ts

```diff
@@ -0,0 +1,176 @@
+import * as actionDraftOrdersTypes from './types/draftOrdersAction.types.js';
+import * as actionTypes from './actionTypes.js';
+import { type AnyAction, combineReducers, type Reducer } from 'redux';
+
+import createReducer, {
+  createReducerWithResult,
+} from '../../helpers/reducerFactory.js';
+import type { DraftOrdersState } from './types/index.js';
+import type { StoreState } from '../../types/storeState.types.js';
+
+export const INITIAL_STATE: DraftOrdersState = {
+  draftOrder: {},
+  allDraftOrders: {},
+  draftOrderCreations: {},
+  updateDraftOrder: {},
+  removeDraftOrder: {},
+};
+
+export const entitiesMapper = {
+  [actionTypes.RESET_DRAFT_ORDERS_STATE]: (
+    state: NonNullable<StoreState['entities']>,
+  ) => {
+    const { draftOrders, ...rest } = state;
+
+    return rest;
+  },
+};
+
+const draftOrderReducer = createReducer(
+  'FETCH_DRAFT_ORDER',
+  { isLoading: false, error: null },
+  actionTypes,
+);
+
+const draftOrder = (state = INITIAL_STATE.draftOrder, action: AnyAction) => {
+  if (actionDraftOrdersTypes.DRAFT_ORDER_ACTIONS.includes(action.type)) {
```

Comment:
This check is redundant because the createReducer already returns to you a reducer that will check for the actions that were specified as its argument appended with _REQUEST, _SUCCESS and _FAILURE.

Reply:
If I remove the type action check, redux doesn't recognize the action passed and the tests break, see the image below, I've even evaluated other sites and they also do the check (https://github.com/Farfetch/blackout/blob/main/packages/redux/src/raffles/reducer.ts#L27)

![Screenshot 2024-01-03 at 11 29 24](https://github.com/Farfetch/blackout/assets/15658297/f5c33481-a23e-49a9-a765-c963c0e1dc0b)

Reply:
If I remove the action type check, createReducer won't recognize the past action and the tests will stop. I've even evaluated other locals and they also do the check (https://github.com/Farfetch/blackout/blob/main/packages/redux/src/raffles/reducer.ts#L27)